Steem 0.18.103

holgern opened the pull request:
https://github.com/steemit/steem-python

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found some lint.

Here's what I've got...


For **recipes/steem**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found it was in an excellent condition.

CJ-Wright commented on recipes/steem/bld.bat:
```diff
@@ -0,0 +1,5 @@
+pip install scrypt==0.8.0
```
please install these via conda-forge

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+    - pkgconfig                    # [unix]
+    # For sha256
+    - openssl 1.0.*                # [unix]
+    - setuptools
```
Are both `setuptools` and `pip` required?

holgern replied:
I will try to remove one

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+    - langdetect
+    - prettytable
+    - voluptuous
+    - sphinx
```
Why are doc building tools build requirements?

holgern replied:
I will try to remove sphinx

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+    - prettytable
+    - voluptuous
+    - sphinx
+    - autopep8
```
Is this a build requirement?

holgern replied:
I will try to remove

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+
+test:
+  imports:
+    - steem
```
Could you please add more import tests?

holgern replied:
I will add all possible combinations

holgern commented:
I use now the scrypt build from conda-forge.

But there are three more packages from pip, which are necessary:
pip install sphinxcontrib-restbuilder
pip install sphinxcontrib-programoutput
pip install pytest-pylint

Should I add them to conda-forge first?

holgern commented:
Ok , added https://github.com/conda-forge/staged-recipes/pull/5109, https://github.com/conda-forge/staged-recipes/pull/5107 and https://github.com/conda-forge/staged-recipes/pull/5106

CJ-Wright commented:
Are these sphinx requirements for building the STEEM docs?

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found some lint.

Here's what I've got...


For **recipes/steem**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found it was in an excellent condition.

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,109 @@
+build:
+  number: 0
+  skip: True  # [py<36]
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
You might try `python -m pip install --no-deps --ignore-installed .`
You will need to add pip to the deps, but then it should ignore all the stated deps.

CJ-Wright replied:
This way you can manage the deps directly without `setuptools` complaining.

holgern replied:
Thanks, I will try

holgern commented:
I have to wait for https://github.com/conda-forge/staged-recipes/pull/5107

holgern commented:
Now i have to wait for https://github.com/conda-forge/staged-recipes/pull/5109

holgern commented:
I seperated build and run requirements.
The package is ready for a second review:).

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,97 @@
+    - steembase.storage
+    - steembase.transactions
+    - steembase.types
+  commands:
```
You might consider testing the CLI as well (just make sure that running the help doesn't blow things up)

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -80,9 +80,11 @@ test:
   commands:
     - conda inspect linkages -p $PREFIX steem  # [not win]
     - conda inspect objects -p $PREFIX steem   # [osx]
+    - python steempy --help
```
Does this require a call to python? https://github.com/steemit/steem-python#cli-tools-bundled

CJ-Wright commented on recipes/steem/meta.yaml:
```diff
@@ -0,0 +1,100 @@
+  number: 0
+  skip: True  # [py<36]
+  script: python -m pip install --no-deps --ignore-installed .
+
```
You need to declare entry points (cli stuff) in the build https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points

holgern replied:
Thanks for the tip.
I added them.

holgern commented:
I have to wait for https://github.com/conda-forge/staged-recipes/pull/5129

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found some lint.

Here's what I've got...


For **recipes/steem**:

 * The build section contained an unexpected subsection name. entrypoints is not a valid subsection name.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/steem```) and found it was in an excellent condition.

CJ-Wright commented:
Do these build on your local machine?

holgern commented:
Problem is that packages are missing. I'm waiting for https://github.com/conda-forge/staged-recipes/pull/5138

CJ-Wright commented:
I guess my point was that you could keep testing for the next req on your local rather than pinging it off the CI.

holgern commented:
Sorry, you are right. I'm doing this now.

holgern commented:
There is a bug in the toml package. It is not found by pkg_resources.require("toml")
I created a pull request https://github.com/conda-forge/toml-feedstock/pull/3, in order to fix this.

holgern commented:
The package can be merged, It does not need toml, with the newest patch. Pipfile and toml are only used within the Docker build, so I removed it. It is also removed in the newest git commit of steem-python.
